pandas/core/algorithms.py

```diff
@@ -931,7 +931,7 @@ def value_counts_internal(
             # For backwards compatibility, we let Index do its normal type
             #  inference, _except_ for if if infers from object to bool.
             idx = Index(keys)
-            if idx.dtype == bool and keys.dtype == object:
+            if idx.dtype in [bool, "string"] and keys.dtype == object:
                 idx = idx.astype(object)
```

Comment:
This is needed for 2.3.x in addition to the changes on main, because here the `Index(..)` is not yet updated to `idx = Index(keys, dtype=keys.dtype)` after the deprecation enforcement